pkg/db/client.go

```diff
@@ -24,6 +24,28 @@ type SqlResult interface {
 	RowsAffected() (int64, error)
 }
 
+type Sqler interface {
+	ToSql() (string, []interface{}, error)
+}
+
+func SqlFmt(format string, a ...any) Sqler {
+	return sqlFmt{
+		format: format,
+		a:      a,
+	}
+}
+
+type sqlFmt struct {
+	format string
+	a      []any
+}
+
+func (s sqlFmt) ToSql() (qry string, args []interface{}, err error) {
+	qry = fmt.Sprintf(s.format, s.a...)
+
+	return
+}
```

Comment:
depending on how nit-picky you like it, you could change all `interface{}` => `any`. Makes it nicer readable, but is optional.

From the funcs name it makes sense to parse the format with the args in `ToSql`. However there is something odd here: You always return an empty args slice as the second return value. Additionally parsing it like this poses to be a security risk when it comes to sql injections imho.

Reply:
This one is for a mix of string formatting and sql placeholder replacement..  
useful if one wants to replace things like the table name which is no possible for sql placeholders..